src/services/createRoute.ts

```diff
@@ -15,17 +15,34 @@ import { checkDuplicateKeys } from '@/utilities/checkDuplicateKeys'
 export function createRoute<
   const TName extends string | undefined = undefined,
   const TPath extends string | Path | undefined = undefined,
-  const TQuery extends string | Query | undefined = undefined
->(options: CreateRouteOptionsWithoutParent<TName, TPath, TQuery>): Route<ToKey<TName>, Host<'', {}>, ToPath<TPath>, ToQuery<TQuery>>
+  const TQuery extends string | Query | undefined = undefined,
+  TComponent extends Component | undefined = undefined
```

Comment:
does this new TComponent generic add something that's useful for us in future PRs? It's not clear to me what retaining the type for component does for our route definition right now

Reply:
Correct this isn't really useful in the PR but it will be in the next one